tests/deploy/test_scheduler_agent.py

```diff
@@ -1286,6 +1285,72 @@ def make_resources(
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
 
+async def test_skipped_for_dependencies_with_event_propagation(agent: TestAgent, make_resource_minimal):
+    """
+    Ensure that a resource that was skipped for its dependencies gets scheduled when they succeed
+    """
+
+    rid1 = ResourceIdStr("test::Resource[agent1,name=1]")
+    rid2 = ResourceIdStr("test::Resource[agent2,name=2]")
+
+    # make both agent's executors managed
+    executor1: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent1", executor1)
+
+    executor2: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent2", executor2)
+
+    resources = {
+        rid1: make_resource_minimal(rid=rid1, values={"value": "r1_value"}, requires=[]),
+        rid2: make_resource_minimal(rid=rid2, values={"value": "r2_value"}, requires=[rid1]),
+    }
+    await agent.scheduler._new_version(version=1, resources=resources, requires=make_requires(resources))
+
+    # We set rid1 to fail so that rid2 has a failing dependency
+    await retry_limited_fast(lambda: rid1 in executor1.deploys)
+    executor1.deploys[rid1].set_result(const.HandlerResourceState.failed)
+
+    # We set rid2 to skip for its dependencies to assert that it gets scheduled when rid1 succeeds
+    await retry_limited_fast(lambda: rid2 in executor2.deploys)
+    executor2.deploys[rid2].set_result(const.HandlerResourceState.skipped_for_dependency)
+    await retry_limited_fast(lambda: agent.executor_manager.executors["agent2"].execute_count == 1)
+
+    assert agent.scheduler._state.resource_state[rid1] == state.ResourceState(
+        status=state.ComplianceStatus.NON_COMPLIANT,
+        deployment_result=state.DeploymentResult.FAILED,
+        blocked=state.BlockedStatus.NO,
+    )
+
+    assert agent.scheduler._state.resource_state[rid2] == state.ResourceState(
+        # We are skipped, so not compliant
+        status=state.ComplianceStatus.NON_COMPLIANT,
+        deployment_result=state.DeploymentResult.SKIPPED,
+        blocked=state.BlockedStatus.TRANSIENT,
+    )
+
+    agent.executor_manager.reset_executor_counters()
+    await agent.scheduler.deploy_resource(rid1, reason="Deploy rid1", priority=TaskPriority.USER_DEPLOY)
+
+    await retry_limited_fast(lambda: rid1 in executor1.deploys)
+    executor1.deploys[rid1].set_result(const.HandlerResourceState.deployed)
+
+    await retry_limited_fast(lambda: rid2 in executor2.deploys)
```

Comment:
I'd add a comment here, because this is the main assertion of this test case
```suggestion
    # verify that rid2 gets scheduled due to the recovery event
    await retry_limited_fast(lambda: rid2 in executor2.deploys)
```